src/Addon.ts

```diff
@@ -0,0 +1,111 @@
+import { Request } from 'express';
+
+import { AuthError, AuthErrorCodes } from './AuthError';
+import { CustomTokenExtractor, Products, TokenPayload } from './types';
+import * as util from './util';
+
+export interface Credentials {
+  sharedSecret: string;
+}
+export type LoadCredentials = (clientKey: string) => Promise<Credentials | undefined>;
+export type SaveCredentials<RequestBody> = (
+  id: string,
+  body: RequestBody,
```

Comment:
As an incompatible improvement for future releases request body here may be replaced with a type with all expected fields defined. Right now the type of this parameter doesn't contribute to the type safety of the module.